source/code/providers/support/runasprovider.h

```diff
@@ -54,6 +56,11 @@ namespace SCXCore
         {
             m_Configurator = configurator;
         }
+        
+        void SetSystemInfo(SCXCoreLib::SCXHandle<SCXSystemLib::SystemInfo> si)
+        {
+        	sysInfo = si;
```

Comment:
First, by convention, all class members are decorated with m_ (for member).

Second, see my PAL comments. I don't see a point of changing SystemInfo to store this. Why not just store it yourself upon ::Load() execution. When you can just store it yourself, why go through the work to make SystemInfo store it with no obvious benefit?